doc/modules/pipeline.rst

```diff
@@ -124,6 +120,36 @@ i.e. if the last estimator is a classifier, the :class:`Pipeline` can be used
 as a classifier. If the last estimator is a transformer, again, so is the
 pipeline.
 
+Caching transformers: avoid to repeat computation
+-------------------------------------------------
+
+.. currentmodule:: sklearn.pipeline
+
+Fitting transformers may sometimes computational expensive. The
+:class:`Pipeline` allows to cache the transformer after calling `fit`.
+This feature is used to avoid to compute the fit transformers within a pipeline
+if the parameters and input data are identical. A typical example is the case of
+a grid search in which the transformers can be fitted only once and reuse for
+each configuration.
+
+The parameter ``memory`` is needed in order to cache the transformers.
+``memory`` can be either a string containing the directory where to cache the
+transfomers or a ``joblib.Memory`` object::
+
+    >>> from tempfile import mkdtemp
+    >>> from sklearn.decomposition import PCA
+    >>> from sklearn.svm import SVC
+    >>> from sklearn.pipeline import Pipeline
+    >>> estimators = [('reduce_dim', PCA()), ('clf', SVC())]
+    >>> pipe = Pipeline(estimators, memory=mkdtemp())
+    >>> pipe # doctest: +NORMALIZE_WHITESPACE, +ELLIPSIS
+    Pipeline(...,
+             steps=[('reduce_dim', PCA(copy=True,...)),
+                    ('clf', SVC(C=1.0,...))])
```

Comment:
Don't we need a warning explaining that estimators will then be cloned and the original estimators won't be modified? I think that you had such a warning box ready to go in an earlier version of the PR.